MetaMask/metamask-extension
# fix (cherry-pick): UX: NetworkList: Ensure setNetworkClientIdForDomain is called for test networks (#25184)

## **Description**

Cuts down the duplication of the NetworkListItem and ensures that `setNetworkClientIdForDomain` is called for test networks so that manually choosing a test network sends a signal to the dapp to change chain.  

PR: #25195
Commit Hash:
[c67aa3f](c67aa3f)

There were no merge conflicts when picking these commits.



[![Open in GitHub

Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/25070?quickstart=1)

## **Related issues**

Fixes: The current Profile Syncing Feature doesn't fully respect Basic
Functionality when turned off. This ensures that the notifications
feature and profile syncing are actually turned off when Basic
Functionality if off.

## **Manual testing steps**

1. Go to a given dapp, connect using Ethereum Mainnet
2. Manually open MetaMask, click the network picker, choose Sepolia
3. See dapp change to sepolia

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
if applicable
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

app, test code being changed).
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.




## **Description**



[![Open in GitHub
Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/25184?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
5.

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**

Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask
Extension Coding
Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
if applicable
guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

app, test code being changed).
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

## **Description**



[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/25260?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
